File: 10813/10813.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    10813.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2023/04/22 19:28:52 by jihoolee          #+#    #+#              #
#    Updated: 2023/04/22 19:34:22 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def main(input_cnt: int, baskets: list):
    for _ in range(input_cnt):
        i, j = map(int, input().split())
        baskets[i - 1], baskets[j - 1] = baskets[j - 1], baskets[i - 1]
    print(*baskets)


if (__name__ == '__main__'):
    basket_cnt, input_cnt = map(int, input().split())
    baskets = [i + 1 for i in range(basket_cnt)]
    main(input_cnt, baskets)

File: 10813/10813.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   10813.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2023/04/22 19:14:37 by jihoolee          #+#    #+#             */
/*   Updated: 2023/04/22 19:28:25 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <vector>
#include <algorithm>

int main(void) {
  int               move_cnt;
  int               basket_cnt;
  std::cin >> basket_cnt >> move_cnt;
  std::vector<int>  basket(basket_cnt);

  for (size_t idx = 0; idx < basket.size(); ++idx)
    basket[idx] = idx + 1;
  for (int move_i = 0; move_i < move_cnt; ++move_i) {
    int i, j;

    std::cin >> i >> j;
    std::swap(basket[i - 1], basket[j - 1]);
  }
  for (std::vector<int>::iterator it = basket.begin();
        it != basket.end();
        ++it)
    std::cout << *it << ' ';
  return 0;
}
